Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1541,17 +1541,23 @@ extension BrowserViewController: URLBarDelegate {
     }
 
     func urlBar(urlBar: URLBarView, didSubmitText text: String) {
-        // If we can't make a valid URL, do a search query.
-        // If we still don't have a valid URL, something is broken. Give up.
         let engine = profile.searchEngines.defaultEngine
-        guard let url = URIFixup.getURL(text) ??
-                        engine.searchURLForQuery(text) else {
+        let url: NSURL
+
+        if let fixupURL = URIFixup.getURL(text) {
+            // The user entered a URL, so use it.
+            url = fixupURL
+        } else if let searchURL = engine.searchURLForQuery(text) {
+            // We couldn't build a URL, so do a search query.
+            url = searchURL
+            Telemetry.recordEvent(SearchTelemetry.makeEvent(engine: engine, source: .URLBar))
+        } else {
+            // We still don't have a valid URL, so something is broken. Give up.
             log.error("Error handling URL entry: \"\(text)\".")
```

Comment:
Yeah, it looks like we use `log.error` in lots of cases that are bad, but out of our control, so we probably wouldn't want to terminate. XCGLogger actually has a higher severity level, `log.severe` which we don't use anywhere --  asserting here would make it equivalent to Android's `log.wtf`.

That, or perhaps a separate method or overloaded implementation of `error` that takes an additional `Bool` for assertions. Anyway, filed [bug 1321331](https://bugzilla.mozilla.org/show_bug.cgi?id=1321331) for this.